ext/dcos-installer/dcos_installer/action_lib/configure.py

```diff
@@ -6,34 +7,45 @@
 import gen
 import pkgpanda
 import gen.installer.bash
-from dcos_installer.util import SERVE_DIR
+from dcos_installer.util import ARTIFACT_DIR, GENCONF_DIR, SERVE_DIR
 
 log = logging.getLogger(__name__)
 
 
 def do_configure(gen_config):
+    # Generate complete config.
     gen_config.update(get_gen_extra_args())
-
-    subprocess.check_output(['mkdir', '-p', SERVE_DIR])
-
     do_validate_gen_config(gen_config)
-
     gen_out = gen.generate(arguments=gen_config)
-    gen.installer.bash.generate(gen_out, SERVE_DIR)
 
-    # Get bootstrap from artifacts
-    fetch_bootstrap(gen_out.arguments['bootstrap_id'])
-    # Write some package metadata
-    pkgpanda.util.write_json('/genconf/cluster_packages.json', gen_out.cluster_packages)
+    # Generate install/upgrade prereqs.
+    subprocess.check_output(['mkdir', '-p', SERVE_DIR])
+    gen.installer.bash.generate(gen_out, SERVE_DIR)
+    fetch_bootstrap(ARTIFACT_DIR, SERVE_DIR, gen_out.arguments['bootstrap_id'])
+    fetch_packages(ARTIFACT_DIR, SERVE_DIR)
+    pkgpanda.util.write_json(os.path.join(GENCONF_DIR, 'cluster_packages.json'), gen_out.cluster_packages)
```

Comment:
I very intentionally wasn't / don't use os.path.join. This is mainly because the behavior in some cases is never what I want:
```
os.path.join("foo", "/bar")
'/bar'
```

Rather I generally make hard assertions about how things must come in. Please switch back to not using os.path.join. Windows support for all this code is not anywhere even near the roadmap (and if it was probably would do a custom join).